### PR TITLE
methods in family service

### PR DESCRIPTION
**Pull request by ayten-github**

### CODE REVIEW

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java [hunk 1]
```diff
@@ -101,31 +108,47 @@ public FamilyDto bornChild(FamilyDto familyDto, String masculineName, String fem
...
+        return index==null||familyList.size()<=index? Optional.empty():
+        Optional.of(familyList.get(index.intValue()));
     }
 
     @Override
-    public List<Pet> getPets(Long index) {
-        return null;
+    public Optional<Pet> getPets(Long index) {
+        List<Family> familyList=familyDao.getAllFamilies();
+        return index>= familyList.size()? Optional.empty():
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/service/FamilyServiceImpl.java]: Why is there no index<0 condition?

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java [hunk 2]
```diff
@@ -101,31 +108,47 @@ public FamilyDto bornChild(FamilyDto familyDto, String masculineName, String fem
 
     @Override
     public Family adoptChild(Family family, Human child) {
-        return null;
+        if(family==null||child==null) return null;
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/service/FamilyServiceImpl.java]: It doesn't make sense to return null, we can handle it easily with Optional

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java [hunk 3]
```diff
@@ -101,31 +108,47 @@ public FamilyDto bornChild(FamilyDto familyDto, String masculineName, String fem
...
     }
 
     @Override
-    public void addPet(Long index, Pet pet) {
-
+    public boolean addPet(final Long index,final Pet pet) {
+        List<Family> familyList=familyDao.getAllFamilies();
+        if (familyList.size()<=index||index<0) return false;
+        familyList.get(index.intValue());
+        return true;
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/service/FamilyServiceImpl.java:149]: this method is not properly written, the goal here was to add a new animal to the family

File: src/main/java/az/edu/strangers/service/FamilyServiceImpl.java [hunk 4]
```diff
@@ -101,31 +108,47 @@ public FamilyDto bornChild(FamilyDto familyDto, String masculineName, String fem
...
-        return null;
+        if(family==null||child==null) return null;
+        family.addChild(child);
+        return family;
     }
 
     @Override
     public void deleteAllChildrenOlderThen(Integer age) {
+        List<Family> familyList=getAllFamilies();
+        familyList.forEach(family -> {
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/service/FamilyServiceImpl.java:118]: int nowYear=LocalDate.now().getYear;

filter(child ->(nowYear-child.getYear())<=age)

**Review by Shammad-Kosayev — APPROVED** [review]
Seems fine

**Pull request merged**